coderd/wsbuilder/wsbuilder.go

```diff
@@ -210,29 +209,18 @@ func (b *Builder) Build(
 	// RepeatableRead isolation ensures that we get a consistent view of the database while
 	// computing the new build.  This simplifies the logic so that we do not need to worry if
 	// later reads are consistent with earlier ones.
-	var err error
-	for retries := 0; retries < 5; retries++ {
-		var workspaceBuild *database.WorkspaceBuild
-		var provisionerJob *database.ProvisionerJob
-		err := store.InTx(func(store database.Store) error {
-			b.store = store
-			workspaceBuild, provisionerJob, err = b.buildTx(authFunc)
-			return err
-		}, &sql.TxOptions{Isolation: sql.LevelRepeatableRead})
-		var pqe *pq.Error
-		if xerrors.As(err, &pqe) {
-			if pqe.Code == "40001" {
-				// serialization error, retry
-				continue
-			}
-		}
-		if err != nil {
-			// Other (hard) error
-			return nil, nil, err
-		}
-		return workspaceBuild, provisionerJob, nil
+	var workspaceBuild *database.WorkspaceBuild
+	var provisionerJob *database.ProvisionerJob
+	err := database.ReadModifyUpdate(store, func(tx database.Store) error {
+		var err error
+		b.store = tx
+		workspaceBuild, provisionerJob, err = b.buildTx(authFunc)
+		return err
+	})
+	if err != nil {
+		return nil, nil, xerrors.Errorf("too many errors; last error: %w", err)
```

Comment:
Not necessarily too many errors -- any error other than serialization error will also cause ReadModifyUpdate to fail.